src/plugins/discover/public/application/main/components/layout/use_discover_histogram.ts

```diff
@@ -6,277 +6,258 @@
-import { useDataState } from '../../hooks/use_data_state';
-import type { DataFetch$, SavedSearchData } from '../../services/discover_data_state_container';
-import type { DiscoverStateContainer } from '../../services/discover_state';
+import { getUiActions } from '../../../../kibana_services';
 import { FetchStatus } from '../../../types';
-import type { DiscoverSearchSessionManager } from '../../services/discover_search_session';
+import { useDataState } from '../../hooks/use_data_state';
 import type { InspectorAdapters } from '../../hooks/use_inspector';
+import type { DataFetch$, SavedSearchData } from '../../services/discover_data_state_container';
 import { checkHitCount, sendErrorTo } from '../../hooks/use_saved_search_messages';
-
-export const CHART_HIDDEN_KEY = 'discover:chartHidden';
-export const HISTOGRAM_HEIGHT_KEY = 'discover:histogramHeight';
-export const HISTOGRAM_BREAKDOWN_FIELD_KEY = 'discover:histogramBreakdownField';
+import { useAppStateSelector } from '../../services/discover_app_state_container';
+import type { DiscoverStateContainer } from '../../services/discover_state';
 
 export interface UseDiscoverHistogramProps {
   stateContainer: DiscoverStateContainer;
   savedSearchData$: SavedSearchData;
   dataView: DataView;
-  savedSearch: SavedSearch;
-  isTimeBased: boolean;
-  isPlainRecord: boolean;
   inspectorAdapters: InspectorAdapters;
-  searchSessionManager: DiscoverSearchSessionManager;
   savedSearchFetch$: DataFetch$;
+  searchSessionId: string | undefined;
 }
 
 export const useDiscoverHistogram = ({
   stateContainer,
   savedSearchData$,
   dataView,
-  savedSearch,
-  isTimeBased,
-  isPlainRecord,
   inspectorAdapters,
-  searchSessionManager,
   savedSearchFetch$,
+  searchSessionId,
 }: UseDiscoverHistogramProps) => {
-  const { storage, data, lens } = useDiscoverServices();
-  const [hideChart, interval, breakdownField] = useAppStateSelector((state) => [
-    state.hideChart,
-    state.interval,
-    state.breakdownField,
-  ]);
+  const services = useDiscoverServices();
+  const timefilter = services.data.query.timefilter.timefilter;
 
   /**
-   * Visualize
+   * API initialization
    */
 
-  const timeField = dataView.timeFieldName && dataView.getFieldByName(dataView.timeFieldName);
-  const [canVisualize, setCanVisualize] = useState(false);
+  const [unifiedHistogram, setUnifiedHistogram] = useState<UnifiedHistogramInitializedApi>();
 
-  useEffect(() => {
-    if (!timeField) {
-      return;
-    }
-    getVisualizeInformation(
-      getUiActions(),
-      timeField,
-      dataView,
-      savedSearch.columns || [],
-      []
-    ).then((info) => {
-      setCanVisualize(Boolean(info));
-    });
-  }, [dataView, savedSearch.columns, timeField]);
-
-  const onEditVisualization = useCallback(
-    (lensAttributes: TypedLensByValueInput['attributes']) => {
-      if (!timeField) {
+  const setUnifiedHistogramApi = useCallback(
+    (api: UnifiedHistogramApi | null) => {
```

Comment:
When and why is this called with `null`?

Reply:
Typically on mount and unmount, although it's not 100% clear to me when/why React passes `null` -- something related to low level hooks behaviour. In any case the `RefCallback<T>` type expects `(instance: T | null): void`, so it needs be able to handle `null`: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bd8f0672368c83af7263c6e28eba54f299a942b5/types/react/index.d.ts#L86.